calico_kubernetes/calico_kubernetes.py

```diff
@@ -724,6 +723,84 @@ def _log_interfaces(namespace):
         # Don't exit if we hit an error logging out the interfaces.
         logger.exception("Ignoring error logging interfaces")
 
+
+def load_config():
+    """
+    Loads configuration for the plugin - returns a dictionary.
+
+    Looks first in environment, then in local config file.
+    """
+    # First, read the config file and get defaults.
+    config = read_config_file()
+
+    # Get config from environment.
+    env_config = {
+            CALICO_IPAM_VAR: os.environ.get(CALICO_IPAM_VAR),
+            KUBE_API_ROOT_VAR: os.environ.get(KUBE_API_ROOT_VAR),
+            DEFAULT_POLICY_VAR: os.environ.get(DEFAULT_POLICY_VAR),
+            KUBE_AUTH_TOKEN_VAR: os.environ.get(KUBE_AUTH_TOKEN_VAR),
+    }
+
+    # Environment variables take precedence.
+    for k,v in env_config.iteritems():
+        if v is not None:
+            logger.debug("Use env variable: %s=%s", k, v)
+            config[k] = v
+
+    # ETCD_AUTHORITY is handled slightly differently - we need to set it in the
+    # environment so that libcalico works correctly.
+    if not ETCD_AUTHORITY_VAR in os.environ:
+        logger.debug("Use env variable: %s=%s",
+                     ETCD_AUTHORITY_VAR,
+                     config[ETCD_AUTHORITY_VAR])
+        os.environ[ETCD_AUTHORITY_VAR] = config[ETCD_AUTHORITY_VAR]
+
+    return config
+
+
+def read_config_file():
+    """
+    Reads the config file on disk and returns configuration dictionary.
+    """
+    # Get the current directory and find path to config file.
+    executable = sys.argv[0]
+    cur_dir = os.path.dirname(executable)
+    config_file = os.path.join(cur_dir, CONFIG_FILENAME)
+
+    # Check that the file exists.
+    if not os.path.exists(config_file):
```

Comment:
To be pedantic, I think you want `os.path.isfile()` here. Unlikely to matter in practice.